specification/authorization/resource-manager/Microsoft.Authorization/preview/2021-11-16-preview/authorization-AccessReviewCalls.json

```diff
@@ -0,0 +1,2665 @@
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PutAccessReviewHistoryDefinition"
+        ],
+        "operationId": "AccessReviewHistoryDefinition_Create",
+        "x-ms-examples": {
+          "PutAccessReviewHistoryDefinition": {
+            "$ref": "./examples/PutAccessReviewHistoryDefinition.json"
+          }
+        },
+        "description": "Create a scheduled or one-time Access Review History Definition",
+        "parameters": [
+          {
+            "name": "historyDefinitionId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The id of the access review history definition."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "AccessReviewHistoryDefinition",
+            "schema": {
+              "$ref": "#/definitions/AccessReviewHistoryDefinition"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "PatchAccessReviewHistoryDefinition"
+        ],
+        "operationId": "AccessReviewHistoryDefinition_GenerateDownloadUri",
+        "x-ms-examples": {
+          "PatchAccessReviewHistoryDefinition": {
+            "$ref": "./examples/PatchAccessReviewHistoryDefinition.json"
+          }
+        },
+        "description": "Generates a uri which can be used to retrieve review history data. This URI has a TTL of 1 day and can be retrieved by fetching the accessReviewHistoryDefinition object.",
+        "parameters": [
+          {
+            "name": "historyDefinitionId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The id of the access review history definition."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "AccessReviewHistoryDefinition",
+            "schema": {
+              "$ref": "#/definitions/AccessReviewHistoryDefinition"
+            }
+          },
+          "default": {
+            "description": "Unexpected error",
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/accessReviewHistoryDefinitions/{historyDefinitionId}/instances": {
```

Comment:
>instances [](http://example.com/codeflow?start=124&length=9)

How are instances created?

Reply:
> > instances
> 
> How are instances created? Seems like you are missing an API similar to: "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/accessReviewScheduleDefinitions/{scheduleDefinitionId}/instances/{id}"

Instances are auto generated. If the definition has schedule settings defined then instances will be created according to the schedule, i.e. once a month or annually. If it is a non-recurring definition, then the definition will only have one instance which will be created immediately. So something like:
accessReviewHistoryDefinitions/{definition-id}/instances is only what we have been using for our Graph API. We could revisit this and add something similar if customers request it but for now it's not needed.